pkgs/development/compilers/gcc/8/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, targetPackages, fetchurl, fetchpatch, noSysDirs
+{ stdenv, targetPackages, fetchurl, fetchFromGitHub, fetchpatch, noSysDirs
```

Comment:
I'm not sure I like the idea of introducing all this very specific stuff in the generic gcc8 expression. At a glance, this looks to me like all these changes can be achieved using overrides? And in that case I think it would be better to have a separate expression file containing these overrides, and using the overridden version to define the stdenv for this platform.

Reply:
Sorry for the delay; it is indeed not trivially clear where this should go — maybe @Ericson2314 and/or @matthewbauer can provide some input, being familiar with stdenv/bootstrapping stuff?

Reply:
Possibly here:
https://github.com/NixOS/nixpkgs/blob/04cea8b44a65a9cbb515b5d1996ccbe80088a84a/pkgs/stdenv/cross/default.nix#L56-L68

Reply:
I am also for putting this behind a different attribute and use overrides.

Reply:
@lheckemann I think it's fine. See the very similar changes for VC4 and Redox for GCC 6. If we could break up the GCC derivation like we do with the LLVM family of projects, we could do better, but we haven't done that yet so I think GCC is just kinda doomed to be messier.